src/native_hooks/all_scripts.hpp

```diff
@@ -121,6 +121,18 @@ namespace big
 				NETWORK::NETWORK_CONCEAL_PLAYER(ctx->get_arg<Player>(0), ctx->get_arg<int>(1), ctx->get_arg<int>(2));
 		}
 
+		void REPORT_CRIME(rage::scrNativeCallContext* src)
+		{
+			Player player = src->get_arg<Player>(0);
+			int crime     = src->get_arg<int>(1);
+			int points    = src->get_arg<int>(2);
+
+			if (g.self.never_wanted && player == self::id)
```

Comment:
What is even the purpose of checking the 0th parameter? Is there ever a scenario where it's called on other players in the scripts?

Reply:
There's only one specific scenario in the `am_taxi` script:
```c
PLAYER::REPORT_CRIME(NETWORK::NETWORK_GET_PLAYER_INDEX(PLAYER::INT_TO_PARTICIPANTINDEX(Local_67.f_2[bParam0 /*26*/].f_3)), 37, PLAYER::GET_WANTED_LEVEL_THRESHOLD(2));
```
This is called if the participant player leaves the taxi without paying for the fare.

Reply:
Does am_taxi do any other checks?

Reply:
Nope. All the other checks are for the local player.